extensions/prototype-content-with-issues-table/app.R

```diff
@@ -0,0 +1,197 @@
+               tag == "build_jupyter" ~ "Build Jupyter Notebook",
+               tag == "packrat_restore" ~ "Packrat Cache Restore",
+               tag == "python_restore" ~ "Python Envrionment Restore",
+               tag == "configure_report" ~ "Parametrized Report Configure",
+               tag == "run_app" ~ "Run R Application",
+               tag == "run_api" ~ "Run Plumber API",
+               tag == "run_tensorflow" ~ "Run Tensorflow API Model",
+               tag == "run_python_api" ~ "Run Python API",
+               tag == "run_dash_app" ~ "Run Dash Application",
+               tag == "run_gradio_app" ~ "Run Gradio Application",
+               tag == "run_streamlit" ~ "Run Streamlit Application",
+               tag == "run_bokeh_app" ~ "Run Bokeh Application",
+               tag == "run_fastapi_app" ~ "Run Python FastAPI",
+               tag == "run_pyshiny_app" ~ "Run PyShiny Application",
+               tag == "render_shiny" ~ "Render Quarto or RMarkdown Interactive Document",
+               tag == "run_voila_app" ~ "Run Voila Application",
+               tag == "run_pyshiny_app" ~ "Run PyShiny Application",
+               tag == "ctrl_extraction" ~ "Extract Parameter Controls From Report",
+               TRUE ~ tag)) |>
+        # map exit codes to something more readable 
+        mutate(exit_code = as.character(exit_code)) |>
+        mutate(exit_code = case_when(
+               exit_code == "1" ~ "Ended with errors",
+               exit_code == "127" ~ "Command could not be executed outside container",
+               exit_code == "137" ~ "Application terminated due to low memory",
+               exit_code == "2" ~ "Command encountered an error during execution",
+               exit_code == "134" ~ "Terminated due to critical error, potential segfault",
+               exit_code == "255" ~ "Abnormal termination of process",
+               exit_code == "130" ~ "Process terminated by user-initiated signal",
+               exit_code == "13" ~ "File permissions issue",
+               exit_code == "15" ~ "Process terminated externally",
+               TRUE ~ exit_code))
+    },
+    error = function(e) {
+      # content item does not have any jobs 
+      NULL
+    }
+  )
+  
+  if (is.null(failed_jobs) || nrow(failed_jobs) == 0) {
+    return(NULL)
+  } else {
+    print(item)
+    print(failed_jobs)
+    # handle content without usage data, such as unpublished content
+    last_visit <- usage %>%
+      filter(content_guid == item$content$guid) %>%
+      slice_max(timestamp) %>%
+      select(user_guid, timestamp)
+    # some content with issues do not have visit data (ex: unpublished)
+    if (nrow(last_visit) == 0){
+      visit_timestamp <- as.POSIXct(0) 
+      visitor_username <- "No visit data exists for this content"
+    } else {
+      visit_timestamp <- last_visit$timestamp
+      visitor_username <- get_user_info(client, last_visit$user_guid, connectServer)
+    }
```

Comment:
Do we need the last visited by? IF we don't add that here would this and the above if branch be much simpler?

Reply:
Last visited can be only the timestamp of the visit, as admins and publishers could always dig deeper into visit data if they notice the failing content was visited. Removing considering the user GUID and resolving the associated username would clean up logic and get rid of get_user_info(), so let's do it.